src/fields/fft_poisson_solver/fft/WrapRocDST.cpp

```diff
@@ -0,0 +1,342 @@
+                                                    std::size_t(expanded_size[1]),
+                                                    std::size_t(expanded_size[2]))};
+
+#ifdef AMREX_USE_FLOAT
+        rocfft_precision precision = rocfft_precision_single;
+#else
+        rocfft_precision precision = rocfft_precision_double;
+#endif
+
+        // Initialize fft_plan.m_plan with the vendor fft plan.
+        rocfft_status result;
+        result = rocfft_plan_create(&(dst_plan.m_plan),
+                                    rocfft_placement_notinplace,
+                                    rocfft_transform_type_real_forward,
+                                    precision,
+                                    dim,
+                                    lengths,
+                                    1,
+                                    nullptr);
+
+        RocFFTUtils::assert_rocfft_status("rocfft_plan_create", result);
+
+        // Store meta-data in dst_plan
+        dst_plan.m_position_array = position_array;
+        dst_plan.m_fourier_array = fourier_array;
+
+        return dst_plan;
+    }
+
+    void DestroyPlan (DSTplan& dst_plan)
+    {
+        rocfft_plan_destroy( dst_plan.m_plan );
+    }
+
+    template<direction d>
+    void Execute (DSTplan& dst_plan){
+        HIPACE_PROFILE("AnyDST::Execute()");
+
+        // Swap position and fourier space based on execute direction
+        amrex::FArrayBox* position_array =
+            (d == direction::forward) ? dst_plan.m_position_array : dst_plan.m_fourier_array;
+        amrex::FArrayBox* fourier_array =
+            (d == direction::forward) ? dst_plan.m_fourier_array : dst_plan.m_position_array;
+
+        // Expand in position space m_position_array -> m_expanded_position_array
+        ExpandR2R(*dst_plan.m_expanded_position_array, *position_array);
+
+        rocfft_status result;
+
+        rocfft_execution_info execinfo = NULL;
+        result = rocfft_execution_info_create(&execinfo);
+        RocFFTUtils::assert_rocfft_status("rocfft_execution_info_create", result);
+
+        std::size_t buffersize = 0;
+        result = rocfft_plan_get_work_buffer_size(dst_plan.m_plan, &buffersize);
+        RocFFTUtils::assert_rocfft_status("rocfft_plan_get_work_buffer_size", result);
+
+        result = rocfft_execution_info_set_stream(execinfo, amrex::Gpu::gpuStream());
+        RocFFTUtils::assert_rocfft_status("rocfft_execution_info_set_stream", result);
+
+        // R2C FFT m_expanded_position_array -> m_expanded_fourier_array
+        // 2nd argument type still wrong, should be void*
+        // reinterpret_cast<AnyFFT::Complex*>(dst_plan.m_expanded_fourier_array->dataPtr()), //3rd arg
+        result = rocfft_execute(dst_plan.m_plan,
+                                (void**)&(dst_plan.m_expanded_position_array),
+                                (void**)&(dst_plan.m_expanded_fourier_array),
+                                execinfo);
```

Comment:
The code compiles well indeed but when trying with this [input file](https://github.com/Hi-PACE/hipace/files/6984119/inputs.txt), I get a segfault in Execute. The segfault disappears when commenting out these lines.

Reply:
Yeah, looking at the comments by @wetzel-desy before these lines, I guess we need to double-check the API contract for `rocfft`.

Reply:
@wetzel-desy do you have any idea of what could go wrong here, and how to fix it?

Reply:
I honestly took inspiration from the WarpX implementation, which calls rocfft_execute. So there might be a difference between the FFT and DST with regard to the underlying data types of the two arrays that are supplied as input. But as I am not involved with the overall structure, I wouldn't know how to fix this without testing... Sorry.

Reply:
Does this work?
```
void* in[2] = {dst_plan.m_expanded_position_array->dataPtr(), nullptr};
void* out[2] = {dst_plan.m_expanded_fourier_array->dataPtr(), nullptr};

result = rocfft_execute(dst_plan.m_plan,
                        in,
                        out,
                        execinfo);
```

Reply:
Excellent, that did it indeed!! With the correct [profile](https://github.com/Hi-PACE/hipace/files/7148161/profile.sh.txt) and [submission](https://github.com/Hi-PACE/hipace/files/7148164/script.sh.txt) script, I unfortunately didn't manage to get openPMD working yet (get a runtime error) but adding some in-situ diags with
```c++
	auto& ptile = m_multi_beam.getBeam(0);
        const auto& aos = ptile.GetArrayOfStructs();
        const auto& pos_structs = aos.begin();
        amrex::Gpu::DeviceVector<amrex::Real> dV(1);
        amrex::Vector<amrex::Real> hV(1);
        auto pdV = dV.begin();
        amrex::ParallelFor(
            ptile.numParticles(),
            [=] AMREX_GPU_DEVICE (long ip) {
		amrex::Real x = pos_structs[ip].pos(0);
                amrex::Gpu::Atomic::Add(pdV, x*x);
            });
        amrex::Gpu::copy(amrex::Gpu::deviceToHost, dV.begin(), dV.end(), hV.begin());
        amrex::Print()<<hV[0]<<'\n';
```
line 400 in Evolve I could check that the evolution was the same on Spock as on Summit. Thank you all!
![Screenshot 2021-09-11 at 18 20 13](https://user-images.githubusercontent.com/26292713/132954500-357f4e4d-ca3f-4341-9dc7-05beb3b3ff6e.png).